mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -1285,14 +1400,53 @@ static ParseResult parseOptionalCallFuncPtr(
   return success();
 }
 
+static ParseResult resolveOpBundleOperands(
+    OpAsmParser &parser, SMLoc loc, OperationState &state,
+    ArrayRef<SmallVector<OpAsmParser::UnresolvedOperand>> opBundleOperands,
+    ArrayRef<SmallVector<Type>> opBundleOperandTypes,
+    StringAttr opBundleSizesAttrName) {
+  assert(opBundleOperands.size() == opBundleOperandTypes.size() &&
+         "operand bundle operand groups and type groups should match");
+
+  unsigned opBundleIndex = 0;
+  for (const auto &[operands, types] :
+       llvm::zip(opBundleOperands, opBundleOperandTypes)) {
```

Comment:
```suggestion
       llvm::zip_equal(opBundleOperands, opBundleOperandTypes)) {
```
nit: zip_equal should assert if the lengths don't match so you may use it to replace the assert above.